src/engine/Match_search_mode.ml

```diff
@@ -564,8 +564,11 @@ let rec filter_ranges (env : env) (xs : (RM.t * MV.bindings list) list)
                    | Xlang.LAliengrep ->
                        raise Impossible
                  in
+                 let ast, _ = Lazy.force env.xtarget.lazy_ast_and_errors in
+                 (* This will incur some cost from the visitor, but it shouldn't be much. *)
```

Comment:
this comment is hard to read because it's not obvious from environment_of_any that internally this runs a visitor.